### ISSUE TITLE
redudant

### ISSUE THREAD

**Issue by nmn** [VSCodeLinter/vscode-linter#1]

**Comment by grotewold** [comment]
You're totally right, but how much code did you right? 100 lines? Really not that much, but the idea is to bring it down to as little as 6 by sharing all the commonality between them.

**Comment by nmn** [comment]
Actually, my linter is just about 50 lines of code right now, and all of that is specific to my own linter. There  is some stuff that I could add, like a simple configuration. But even then in most cases, the configuration will be specific to my own linter.

What common parts do you want to extract out? The only one I can think of is something that would conditionally load the linter only if a config file can be found in the project root. So for example, eslint would only be active in a project which has the .eslintrc file in it. I can't think of any other parts of a linter that could be shared.

**Comment by grotewold** [comment]
Common parts, to start, are configuration management, utility functions like quickly enabling/disabling any linter that uses the framework, rudimentary things that every linter parsing through the linters output.

I don't disagree that VSCode is different from Atom or SublimeText due to some built-in language workers, but it still has commonality that could be shared.

**Comment by grotewold** [comment]
Would like to hear @seanmcbreen's thoughts on the issue?

**Comment by seanmcbreen** [comment]
Making a linter as simple to write as possible is definitely the goal - that said we should look at add as many common code-paths into the worker API.  Many of the elements discussed above do have helper support.   For most cases a linter should be in the 50 LOC region.

Things like a helper library to easily map different result patterns into a problems would be cool (in my spell checker most of the code is working out the location of the detected errors, working with my custom config file, etc).

I'd recommend starting with writing one and seeing the gaps you think we have - raise those as issues and we can work out the best next step.  Something like this should be easy... https://github.com/CSSLint/csslint

**Comment by seanmcbreen** [comment]
Oh also please don't post any code you create publicly yet.
